packages/console/src/pages/Roles/components/CreateRoleForm/Footer.tsx

```diff
@@ -0,0 +1,121 @@
+import { type RoleResponse, RoleType, ReservedPlanId } from '@logto/schemas';
+import { useContext } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import ContactUsPhraseLink from '@/components/ContactUsPhraseLink';
+import PlanName from '@/components/PlanName';
+import QuotaGuardFooter from '@/components/QuotaGuardFooter';
+import { isCloud, isDevFeaturesEnabled } from '@/consts/env';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import Button from '@/ds-components/Button';
+import useSubscriptionPlan from '@/hooks/use-subscription-plan';
+import { hasReachedQuotaLimit } from '@/utils/quota';
+import { buildUrl } from '@/utils/url';
+
+type Props = {
+  roleType: RoleType;
+  selectedScopesCount: number;
+  isCreating: boolean;
+  onClickCreate: () => void;
+};
+
+function Footer({ roleType, selectedScopesCount, isCreating, onClickCreate }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { currentTenantId } = useContext(TenantsContext);
+  const { data: currentPlan } = useSubscriptionPlan(currentTenantId);
+  const { data: [, roleCount] = [] } = useSWR<[RoleResponse[], number]>(
+    isCloud &&
+      buildUrl('api/roles', {
+        page: String(1),
+        page_size: String(1),
+        type: roleType,
+      })
+  );
+
+  const isRoleReachLimit = hasReachedQuotaLimit({
```

Comment:
nit: Rename the variable to something like `hasRoleReachedLimit` or `isRoleLimitReached`

Reply:
Updated.